### PR TITLE
Case 20410 - Automatic Content Archives doesn't restore correctly with old Tutorial content set

### PR DESCRIPTION
**Pull request by roxanneskelly**
The tutorial content set had a content set Id of Null, which caused a failure
to parse the content set.

### CODE REVIEW

**Comment by roxanneskelly** [comment]
https://highfidelity.manuscript.com/f/cases/20410/Automatic-Content-Archives-doesn-t-restore-correctly-with-old-Tutorial-content-set

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18537/HighFidelity-Beta-PR15308-cf9fe00.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18537/HighFidelity-Beta-Interface-PR15308-cf9fe00.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18537/HighFidelity-Beta-PR15308-cf9fe00.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18537/HighFidelity-Quest-Beta-PR15308-cf9fe00.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18537/HighFidelity-Beta-PR15308-cf9fe00.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18537/HighFidelity-Beta-Interface-PR15308-cf9fe00.exe

File: libraries/octree/src/OctreeEntitiesFileParser.cpp [hunk 1]
```diff
@@ -107,14 +107,23 @@ bool OctreeEntitiesFileParser::parseEntities(QVariantMap& parsedEntities) {
...
                 return false;
             }
+
+            if (idString == "{00000000-0000-0000-0000-000000000000}") {
+                // some older archives may have a null string id, so 
+                // return success without setting parsedEntities, 
+                // which will result in a new uuid for the restored
+                // archive.  (not parsing and using isNull as parsing
+                // results in null if there is a corrupt string)
+                return true;
```
Comment (SimonWalton-HiFi) [inline libraries/octree/src/OctreeEntitiesFileParser.cpp]: The Entities in the file will still get parsed eventually, right?

Comment (roxanneskelly) [reply libraries/octree/src/OctreeEntitiesFileParser.cpp]: ah, didn't catch that as the Id was the last thing in the entity file.  WIll fix.

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18543/HighFidelity-Beta-PR15308-ac97a82.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18543/HighFidelity-Quest-Beta-PR15308-ac97a82.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18543/HighFidelity-Beta-PR15308-ac97a82.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18543/HighFidelity-Beta-Interface-PR15308-ac97a82.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18543/HighFidelity-Beta-PR15308-ac97a82.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18543/HighFidelity-Beta-Interface-PR15308-ac97a82.exe

File: libraries/octree/src/OctreeEntitiesFileParser.cpp [hunk 2]
```diff
@@ -107,14 +107,21 @@ bool OctreeEntitiesFileParser::parseEntities(QVariantMap& parsedEntities) {
...
 
-            parsedEntities["Id"] = idValue;
-            gotId = true;
+            // some older archives may have a null string id, so
+            // return success without setting parsedEntities,
+            // which will result in a new uuid for the restored
+            // archive.  (not parsing and using isNull as parsing
+            // results in null if there is a corrupt string)
+
+            if (idString != "{00000000-0000-0000-0000-000000000000}") {
```
Comment (davidkelly) [inline libraries/octree/src/OctreeEntitiesFileParser.cpp]: you can also use QUuid::fromString() and .isNull() - that way with/without braces would be safe.

**Pull request merged**